crates/oxc_linter/src/rules/jest/consistent_test_it.rs

```diff
@@ -0,0 +1,916 @@
+    OxcDiagnostic::warn(format!(
+        "{x0}(consistent-test-it): Enforce `test` and `it` usage conventions",
+    ))
+    .with_help(format!("Prefer using '{x1:?}' instead of '{x2:?}' within describe"))
+    .with_label(span0)
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+enum TestCaseName {
+    Fit,
+    IT,
+    Test,
+    Xit,
+    Xtest,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct ConsistentTestIt(Box<ConsistentTestItConfig>);
+
+#[derive(Debug, Clone)]
+pub struct ConsistentTestItConfig {
+    within_describe: TestCaseName,
+    within_fn: TestCaseName,
+}
+
+impl std::ops::Deref for ConsistentTestIt {
+    type Target = ConsistentTestItConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for ConsistentTestItConfig {
+    fn default() -> Self {
+        Self { within_describe: TestCaseName::IT, within_fn: TestCaseName::Test }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Jest allows you to choose how you want to define your tests, using the `it` or
+    /// the `test` keywords, with multiple permutations for each:
+    ///
+    /// - **it:** `it`, `xit`, `fit`, `it.only`, `it.skip`.
+    /// - **test:** `test`, `xtest`, `test.only`, `test.skip`.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// /*eslint jest/consistent-test-it: ["error", {"fn": "test"}]*/
+    /// test('foo'); // valid
+    /// test.only('foo'); // valid
+    ///
+    /// it('foo'); // invalid
+    /// it.only('foo'); // invalid
+    /// ```
+    ///
+    /// ```javascript
+    /// /*eslint jest/consistent-test-it: ["error", {"fn": "it"}]*/
+    /// it('foo'); // valid
+    /// it.only('foo'); // valid
+    /// test('foo'); // invalid
+    /// test.only('foo'); // invalid
+    /// ```
+    ///
+    /// ```javascript
+    /// /*eslint jest/consistent-test-it: ["error", {"fn": "it", "withinDescribe": "test"}]*/
+    /// it('foo'); // valid
+    /// describe('foo', function () {
+    ///     test('bar'); // valid
+    /// });
+    ///
+    /// test('foo'); // invalid
+    /// describe('foo', function () {
+    ///     it('bar'); // invalid
+    /// });
+    /// ```
+    ///
+    /// #### Options
+    ///
+    /// This rule can be configured as follows
+    /// ```json5
+    /// {
+    ///     type: 'object',
+    ///     properties: {
+    ///         fn: {
+    ///             enum: ['it', 'test'],
+    ///         },
+    ///         withinDescribe: {
+    ///             enum: ['it', 'test'],
+    ///         },
+    ///     },
+    ///     additionalProperties: false,
+    /// }
+    /// ```
+    ///
+    /// ##### fn
+    /// Decides whether to use `test` or `it`.
+    ///
+    /// ##### withinDescribe
+    /// Decides whether to use `test` or `it` within a `describe` scope.
+    ///
+    ///
+    /// This rule is compatible with [eslint-plugin-vitest](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/consistent-test-it.md),
```

Comment:
We also need to add the rule name here: https://github.com/oxc-project/oxc/blob/328445b4ca821056ea01c19548f784d8df76a5d2/crates/oxc_linter/src/utils/mod.rs#L18-L20. (this a bit unobvious)